pyproject.toml

```diff
@@ -1,2 +1,8 @@
 [build-system]
-requires = ["setuptools", "wheel", "Cython", "numpy", "scipy"]
+requires = [
+    "setuptools",
+    "wheel",
+    "cython>=0.29.20",
+    "numpy>=1.16.6,<1.20",
```

Comment:
Doesn't this exclude numpy  1.20?

Reply:
Yes, but only from the system doing the building.  Let's say numpy 1.20 is installed, then a user does `pip install qutip` - `pip` will recognise that the numpy requirement is not correctly satisfied for the build, but it is for installation, so it'll temporarily install `numpy==1.19.2` (probably) into a venv, do the build within that, then drop `qutip` into the main environment

Reply:
cf https://github.com/qutip/qutip/pull/1429/files#r582891252